File: gachisa-backend/src/main/java/com/gachisa/payment/client/dto/PgPaymentQueryResult.java
package com.gachisa.payment.client.dto;

import com.gachisa.payment.entity.PaymentMethod;

public record PgPaymentQueryResult(
        String paymentKey,
        String pgOrderId,
        int amount,
        String status,
        PaymentMethod paymentMethod,
        String cancellationTransactionKey,
        String cancellationReason,
        int cancelledAmount
) {
}

File: gachisa-backend/src/main/kotlin/com/gachisa/payment/client/dto/PgPaymentQueryResult.kt
package com.gachisa.payment.client.dto

import com.gachisa.payment.entity.PaymentMethod

data class PgPaymentQueryResult(
    val paymentKey: String,
    val pgOrderId: String,
    val amount: Int,
    val status: String,
    val paymentMethod: PaymentMethod?,
    val cancellationTransactionKey: String?,
    val cancellationReason: String?,
    val cancelledAmount: Int,
)
